docs/framework/react/guide/not-found-errors.md

```diff
@@ -4,11 +4,145 @@ title: Not Found Errors
+- ["root" mode](#notfoundmode-root): All not-found errors will be handled by the root route's `notFoundComponent`, regardless of the nearest matching route.
+
+### `notFoundMode: 'fuzzy'`
+
+By default, the router's `notFoundMode` is set to `fuzzy`, which indicates that if a pathname doesn't match any known route, the router will attempt to use the closest matching route with children/(an outlet) and a configured not found component.
+
+> **❓ Why is this the default?** Fuzzy matching to preserve as much parent layout as possible for the user gives them more context to navigate to a useful location based on where they thought they would arrive.
+
+The nearest suitable route is found using the following criteria:
+
+- The route must have children and therefore an `Outlet` to render the `notFoundComponent`
+- The route must have a `notFoundComponent` configured or the router must have a `defaultNotFoundComponent` configured
+
+For example, consider the following route tree:
+
+- `__root__` (has a `notFoundComponent` configured)
+  - `posts` (has a `notFoundComponent` configured)
+    - `$postId` (has a `notFoundComponent` configured)
+
+If provided the path of `/posts/1/edit`, the following component structure will be rendered:
+
+- `<Root>`
+  - `<Posts>`
+    - `<postRoute.notFoundComponent>`
+
+The `notFoundComponent` of the `posts` route will be rendered because it is the **nearest suitable parent route with children (and therefore an outlet) and a `notFoundComponent` configured**.
+
+### `notFoundMode: 'root'`
+
+When `notFoundMode` is set to `root`, all not-found errors will be handled by the root route's `notFoundComponent` instead of bubbling up from the nearest fuzzy-matched route.
+
+For example, consider the following route tree:
+
+- `__root__` (has a `notFoundComponent` configured)
+  - `posts` (has a `notFoundComponent` configured)
+    - `$postId` (has a `notFoundComponent` configured)
+
+If provided the path of `/posts/1/edit`, the following component structure will be rendered:
+
+- `<Root>`
+  - `<Root.notFoundComponent>`
+
+The `notFoundComponent` of the `__root__` route will be rendered because the `notFoundMode` is set to `root`.
+
+## Configuring a route's `notFoundComponent`
+
+To handle both types of not-found errors, you can attach a `notFoundComponent` to a route. This component will be rendered when a not-found error is thrown.
+
+For example, configuring a `notFoundComponent` for a `/settings` route to handle non-existing settings pages:
+
+```tsx
+export const Route = createFileRoute('/settings')({
+  component: () => {
+    return (
+      <div>
+        <p>Settings page</p>
+        <Outlet />
+      </div>
+    )
+  },
+  notFoundComponent: () => {
+    return <p>This setting page doesn't exist!</p>
+  },
+})
+```
+
+Or configuring a `notFoundComponent` for a `/posts/$postId` route to handle posts that don't exist:
```

Comment:
doesn't this contract the above?

> The `notFoundComponent` of the `posts` route will be rendered because it is the **nearest suitable parent route with children (and therefore an outlet) and a `notFoundComponent` configured**.

Reply:
Yes it does. I’ll fix that.